File: Practice problem/Collatz conjecture.cpp
// 연습문제 콜라츠 추측 Level 1

using namespace std;

int solution(int num) {
    long long N = num;
    int answer = 0;
    while(N != 1){
        if(N % 2 == 0){N /= 2;}
        else{N = N*3 + 1;}
        answer++;
        if(answer == 500) {
            answer = -1;
            break;
        }
    }
    return answer;
}

File: Practice problem/Collatz conjecture/Collatz conjecture.py
# 연습문제 콜라츠 추측 Level 1

def solution(num):
    answer = 0
    while answer <= 500:
        if num == 1:
            return answer
        num = int(num / 2) if num % 2 == 0 else num*3+1
        answer += 1
    return -1
